californium-core/src/main/java/org/eclipse/californium/core/network/deduplication/CropRotation.java

```diff
@@ -48,12 +50,12 @@
 	private boolean running = false;
 	private ScheduledExecutorService executor;
 
-	private ExchangeMap[] maps;
-	private int first;
-	private int second;
+	private final ExchangeMap maps[];
+	private volatile int first;
+	private volatile int second;
```

Comment:
Write-access to `first` and `second` is synchronized, while `read` not.
As far as I know, this doesn't ensure, the `read` gets the right values.
Therefore I added `volatile`. Alternatively the read could be done also `synchronized`, but

> 		Exchange prev = maps[f].get(key);
> 		if (prev != null || f == s)
> 			return prev;
> 		prev = maps[s].get(key);
> 

the compare `f == s` indicates, that a none `synchronized` solution was intended.